### ISSUE TITLE
Calling this.swPush.unsubscribe() throws an error

### ISSUE THREAD

**Issue by GeorgeTailor** [angular/angular#24095]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior

After calling 
```
this.swPush.requestSubscription({
            serverPublicKey: this.VAPID_PUBLIC_KEY
        })
```
I am trying to call 
`this.swPush.unsubscribe();`
The above call results in the following error: 

> Error: Uncaught (in promise): Error: Not subscribed to push notifications.

However, I am not receiving the notifications anymore.

## Expected behavior

Calling `this.swPush.unsubscribe();` should successfully unsubscribe user from receiving web push notifications and not throw any errors.

## Minimal reproduction of the problem with instructions
Not able to reproduce due to stackblitz limitations apparently.

https://stackblitz.com/edit/angular-gitter-8i4u4d

throws 

> Uncaught (in promise): NotAllowedError: Registration failed - permission denied

## What is the motivation / use case for changing the behavior?
Confusing behaviour which makes the dev do some hackz in order not to print error to console.

## Environment

<pre><code>
Angular version: 5.2.9

Browser:
- [X] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Platform:

**Comment by gkalpak** [comment]
It is hard to investigate without a reproduction (and more detailed instructions). For example, are you calling `unsubscribe()` after the promise returned from `requestSubscription()` is resolved?

**Comment by GeorgeTailor** [comment]
Yes,  after the promise returned from requestSubscription() is resolved.

**Comment by GeorgeTailor** [comment]
From app dev perspective I can tell that it is really hard to debug minified code.
`requestSubscription()` resolves to:
```
{
    endpoint:"https://fcm.googleapis.com/fcm/send/#code#",
    expirationTime:null,
    options:PushSubscriptionOptions {userVisibleOnly: true, applicationServerKey: ArrayBuffer(65)}
}
```

`await (await navigator.serviceWorker.getRegistration()).pushManager.getSubscription()` resolves to the exactly the same as the above object.

**Comment by gkalpak** [comment]
Then I really don't know :grin:
Someone needs to debug this and it is impossible without a reproduction.

**Comment by GeorgeTailor** [comment]
I have updated the stackblitz demo. To reproduce the issue you can download the zip, `npm install`, add `serviceWorker: true`, `ng build --prod -aot=false`, and run `http-server` from dist folder. This fresh installation gives the same error on Chrome 66.

**Comment by gkalpak** [comment]
Thx for the repro. Here is the fix: #24162
